lib/admin-page.php

```diff
@@ -304,33 +304,59 @@ function classicpress_check_can_migrate() {
 	);
 	echo '<table id="cp-preflight-checks">' . "\n";
 
+	$disable_wp_version_check = apply_filters( 'classicpress_ignore_wp_version', false );
+
 	// Check: Supported WP version
 	// More versions can be added if they pass the plugin's automated tests.
-	global $wp_version;
-	$wp_version_min = '4.9.0';
-	$wp_version_max = '4.9.8';
-	if (
-		// Version is outside of our "stable release" range...
-		(
-			version_compare( $wp_version, $wp_version_min, 'lt' ) ||
-			version_compare( $wp_version, $wp_version_max, 'gt' )
-		) &&
-		// ... and it's not a known development release
-		! preg_match( '#^4\.9\.9-(alpha|beta)\b#', $wp_version )
-	) {
-		$preflight_checks['wp_version'] = false;
-		echo "<tr>\n<td>$icon_preflight_fail</td>\n<td>\n";
-	} else {
+	
+	if( !empty( $disable_wp_version_check ) ) { // Skip the test and issue a warning
+		$preflight_check_wp_version_status = $icon_preflight_warn;
 		$preflight_checks['wp_version'] = true;
-		echo "<tr>\n<td>$icon_preflight_pass</td>\n<td>\n";
+		
+	} else { // Run the check
+		global $wp_version;
+		
+		$wp_version_min = '4.9.0';
+		$wp_version_max = '4.9.8';
+		
+		// Default values - skips one else
```

Comment:
When you see a code comment like this, you know things are starting to get a bit convoluted :)  Here is the way I think we should structure this code instead:

`if ( version check failed ) {
    if ( apply_filters( 'classicpress_ignore_wp_version', false ) ) {
        // Show warning but allow migration to proceed
    } else {
        // Block migration, show error message
    }
} else {
    // Everything OK, and we don't need to worry about the value of the new filter
}
```